osh.spec

```diff
@@ -225,8 +225,15 @@ fi
 %systemd_preun osh-worker.service
 
 %postun worker
-# osh-worker does not implement reload and restart would interrupt running tasks
-%systemd_postun osh-worker.service
+%if 0%{?fedora} || 0%{?rhel} > 9
+%systemd_postun_with_reload osh-worker.service
+%else
+# Needs systemd 248 or newer
+if [ $1 -ge 1 ] && [ -x "/usr/lib/systemd/systemd-update-helper" ]; then
+    # Package upgrade, not uninstall
+    systemctl set-property osh-worker.service Markers=+needs-reload || :
```

Comment:
It does not work as expected.  I tried it on a staging instance and it **restarted** the worker instead of reloading:
```
Mar 28 15:23:27 osh-test-worker-001.osh-001.preprod.iad2.dc.redhat.com systemd[1]: Queuing reload/restart jobs for marked units…
Mar 28 15:23:27 osh-test-worker-001.osh-001.preprod.iad2.dc.redhat.com systemd[1]: Stopping OpenScanHub worker daemon...
Mar 28 15:23:28 osh-test-worker-001.osh-001.preprod.iad2.dc.redhat.com systemd[1]: osh-worker.service: Deactivated successfully.
Mar 28 15:23:28 osh-test-worker-001.osh-001.preprod.iad2.dc.redhat.com systemd[1]: Stopped OpenScanHub worker daemon.
Mar 28 15:23:28 osh-test-worker-001.osh-001.preprod.iad2.dc.redhat.com systemd[1]: osh-worker.service: Consumed 1h 5min 39.936s CPU time.
Mar 28 15:23:28 osh-test-worker-001.osh-001.preprod.iad2.dc.redhat.com systemd[1]: Starting OpenScanHub worker daemon...
Mar 28 15:23:28 osh-test-worker-001.osh-001.preprod.iad2.dc.redhat.com systemd[1]: Started OpenScanHub worker daemon.
```

Reply:
Hmm...  Either `+needs-reload` marker is not processed correctly on RHEL 9, or `systemd` has not reloaded its own configuration using `daemon-reload` yet so it does not know yet about the added `ExecReload:` line in the unit file.  Internally, the `/usr/lib/systemd/systemd-update-helper` script calls `systemctl try-reload-or-restart` after the update is finished:
```
try-reload-or restart PATTERN...
           Reload one or more units if they support it. If not, stop and then start them instead. This does nothing if the units are not running.
```
so that explains the restart but I would expect this script to call `daemon-reload` first before processing marked units.

@kdudka Could you please try calling `systemctl daemon-reload` and then reinstalling the current `osh-worker` package to see it gets then reloaded correctly?

Reply:
I think that @jamacku looked already and submitted https://github.com/redhat-plumbers/systemd-rhel9/pull/250 based on this.  But to me it seems like an overkill.  Cannot we just invoke `systemctl reload osh-worker.service` in the RPM scriptlet on RHEL-9?

Reply:
@lzaoral, I have just noticed that RHEL8 is running `v239`

Reply:
> systemctl reload should be called after systemctl daemon-reload to use the latest unit file.

I would not worry about that.  The unit file is changed approx. twice a year and the `reload` operation would be useful even if it ran on the outdated version of the unit file.

Reply:
Addressed in e210c3f7581836129341f5a37f6561a6f23a2c8a.